### PR TITLE
Feat/check lbpair v1 existence

### PR DESCRIPTION
**Pull request by 00xSam**
Context: https://discord.com/channels/931167763907612703/953101077232893992/1238204884042383380

### CODE REVIEW

File: ts-client/src/dlmm/index.ts [hunk 1]
```diff
@@ -1084,6 +1101,12 @@ export class DLMM {
     );
     const program = new Program(IDL, LBCLMM_PROGRAM_IDS[opt.cluster], provider);
 
+    if (
+      !this.checkPoolExists(connection, tokenX, tokenY, binStep, baseFactor)
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:1105]: could you renaming function to: isPoolNotExisted ?

Comment (00xSam) [reply ts-client/src/dlmm/index.ts]: renamed to `isPoolExistsed` 👍

File: ts-client/src/dlmm/index.ts [hunk 2]
```diff
@@ -1095,6 +1112,10 @@ export class DLMM {
     );
     const program = new Program(IDL, LBCLMM_PROGRAM_IDS[opt.cluster], provider);
 
+    if (!this.isPoolExisted(connection, tokenX, tokenY, binStep, baseFactor)) {
```
Comment (andrewsource147) [inline ts-client/src/dlmm/index.ts:1115]: why not isPoolExisted return Error Pool already exists?

Comment (00xSam) [reply ts-client/src/dlmm/index.ts]: ohh sorry.

File: ts-client/src/dlmm/index.ts [hunk 3]
```diff
@@ -184,15 +185,31 @@ export class DLMM {
...
         tokenX,
         tokenY,
         binStep,
         baseFactor,
         program.programId
       );
-      await program.account.lbPair.fetch(lbPairKey);
-      return true;
+      const account2 = await program.account.lbPair.fetchNullable(lbPair2Key);
+      if (account2) return lbPair2Key;
```
Comment (codewithgun) [inline ts-client/src/dlmm/index.ts:196]: Can we return true? When account doesn't exist it return false, let's make it consistent

Comment (00xSam) [reply ts-client/src/dlmm/index.ts:196]: Can we do `PublicKey | null`? I need to return the pool address for the warning of exists pool redirect link

**Review by codewithgun — APPROVED** [review]

**Pull request merged**